### PR TITLE
WIP - Istio 1 1

### PR DESCRIPTION
**Pull request by francoispradeau**
WIP - changes to bring istio api 1.1 in. To coordinate with aspenmesh/istio-client-go#17 as it needs to bring in a 1.1-ready istio-client-go

### CODE REVIEW

**Comment by francoispradeau** [comment]
Additional comment - this build will fail until we merge istio-client-ui and have an official version using Istio api 1.1

File: pkg/vetter/util/util.coverprofile [hunk 1]
```diff
@@ -0,0 +1,176 @@
+mode: atomic
```
Comment (m-eaton) [inline pkg/vetter/util/util.coverprofile]: Remove the `.coverprofile` files

File: pkg/vetter/util/istio-inject.go [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+func includeInboundPorts(containers []corev1.Container) string {
+	// Include the ports from all containers in the deployment.
+	return getContainerPorts(containers, func(corev1.Container) bool { return true })
+}
+
+func kubevirtInterfaces(s string) string {
+	return s
+}
+
+func toJSON(m map[string]string) string {
```
Comment (m-eaton) [inline pkg/vetter/util/istio-inject.go]: Double check that both these and `ApplyJSON` are needed.

File: pkg/vetter/util/testdata/0.8/0.8-mesh-config.yaml [hunk 3]
```diff
@@ -87,7 +87,7 @@ metadata:
   creationTimestamp: null
   labels:
     app: istio
-    chart: istio-0.8.0
+    chart: istio-0.8.
```
Comment (m-eaton) [inline pkg/vetter/util/testdata/0.8/0.8-mesh-config.yaml]: Add the 0 back

File: pkg/vetter/util/util.go [hunk 4]
```diff
@@ -73,27 +69,8 @@ var istioInjectNamespaceLabel = map[string]string{
 
 // Following types are taken from
 // https://github.com/istio/istio/blob/master/pilot/pkg/kube/inject/inject.go
+// The needed contents from that file are in istio-ingect.go
```
Comment (m-eaton) [inline pkg/vetter/util/util.go]: `ingect` -> `inject`

File: pkg/vetter/util/util.go [hunk 5]
```diff
@@ -211,7 +172,7 @@ func GetMeshConfig(cm *corev1.ConfigMap) (*meshv1alpha1.MeshConfig, error) {
 		return nil, nil
 	}
 	var cfg meshv1alpha1.MeshConfig
-	if err := ApplyYAML(c, &cfg); err != nil {
+	if err := ApplyYAML(c, &cfg, false); err != nil {
```
Comment (m-eaton) [inline pkg/vetter/util/util.go]: Double check that we need both `ApplyYAML` and the other YAML functions in the inject utils

File: pkg/vetter/util/istio-inject.go (hunk 2 above)
Comment (francoispradeau) [reply pkg/vetter/util/istio-inject.go]: They are fairly different (arguments, and parsing methods) so let's leave them both for now...

**Pull request merged**